Sources/StreamChatUI/ChatMessageList/ChatMessage/GiftPacketSentBubble.swift

```diff
@@ -0,0 +1,264 @@
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        self.selectionStyle = .none
+        self.backgroundColor = .clear
+        setupUI()
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    private func setupUI() {
+        let gesture = UITapGestureRecognizer(target: self, action: #selector(didTap(tap:)))
+        viewContainer = UIView()
+        viewContainer.translatesAutoresizingMaskIntoConstraints = false
+        viewContainer.backgroundColor = .clear
+        viewContainer.clipsToBounds = true
+        viewContainer.addGestureRecognizer(gesture)
+        contentView.addSubview(viewContainer)
+        NSLayoutConstraint.activate([
+            viewContainer.topAnchor.constraint(equalTo: self.contentView.topAnchor, constant: 0),
+            viewContainer.bottomAnchor.constraint(equalTo: self.contentView.bottomAnchor, constant: -Constants.MessageTopPadding)
+        ])
+
+        subContainer = UIView()
+        subContainer.translatesAutoresizingMaskIntoConstraints = false
+        subContainer.backgroundColor = Appearance.default.colorPalette.background6
+        subContainer.layer.cornerRadius = 12
+        subContainer.clipsToBounds = true
+        viewContainer.addSubview(subContainer)
+        NSLayoutConstraint.activate([
+            subContainer.bottomAnchor.constraint(equalTo: viewContainer.bottomAnchor, constant: 0),
+            subContainer.leadingAnchor.constraint(equalTo: viewContainer.leadingAnchor, constant: 0),
+            subContainer.trailingAnchor.constraint(equalTo: viewContainer.trailingAnchor, constant: 0),
+        ])
+    }
+
+    func configureCell(isSender: Bool) {
+        self.isSender = isSender
+        if isSender {
+            viewContainer.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor, constant: cellWidth).isActive = true
+            viewContainer.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor, constant: -8).isActive = true
+        } else {
+            viewContainer.leadingAnchor.constraint(equalTo: self.contentView.leadingAnchor, constant: 8).isActive = true
+            viewContainer.trailingAnchor.constraint(equalTo: self.contentView.trailingAnchor, constant: -cellWidth).isActive = true
+        }
+
+        setFlair()
+        titleLabel = createTitleLabel()
+        amountLabel = createAmountLabel()
+        detailLabel = createDetailsLabel()
+        timestampLabel = createTimestampLabel()
+
+        detailsStack = UIStackView(arrangedSubviews: [titleLabel, amountLabel])
+        detailsStack.axis = .horizontal
+        detailsStack.distribution = .equalSpacing
+        detailsStack.spacing = 2
+        subContainer.addSubview(detailsStack)
+        detailsStack.transform = .mirrorY
+        detailsStack.alignment = .leading
+        detailsStack.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
```

Comment:
@tunguyen-zien  Please check this file for your reference. and update GiftPacket chat bubble accordingly.
https://github.com/timeless-space/stream-chat-swift/blob/bugfix/refactor-chat-bubble/Sources/StreamChatUI/ChatMessageList/CryptoReceiveBubble.swift